cpp/src/arrow/flight/sql/odbc/flight_sql/flight_sql_statement_get_tables.cc

```diff
@@ -0,0 +1,181 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow/flight/sql/odbc/flight_sql/flight_sql_statement_get_tables.h"
+#include "arrow/flight/api.h"
+#include "arrow/flight/sql/odbc/flight_sql/flight_sql_result_set.h"
+#include "arrow/flight/sql/odbc/flight_sql/record_batch_transformer.h"
+#include "arrow/flight/sql/odbc/flight_sql/utils.h"
+#include "arrow/flight/sql/odbc/odbcabstraction/include/odbcabstraction/platform.h"
+#include "arrow/flight/types.h"
+
+namespace driver {
+namespace flight_sql {
+
+using arrow::Result;
+using arrow::flight::FlightClientOptions;
+using arrow::flight::FlightInfo;
+using arrow::flight::sql::FlightSqlClient;
+
+void ParseTableTypes(const std::string& table_type,
+                     std::vector<std::string>& table_types) {
+  bool encountered = false;  // for checking if there is a single quote
+  std::string curr_parse;    // the current string
+
+  for (char temp : table_type) {  // while still in the string
+    switch (temp) {               // switch depending on the character
```

Comment:
Many of these comments are just noise and aren't necessary. I would rather we have docstrings since the purpose of many functions is not necessarily clear. Another thing is that for functions which parse a value, it would be more helpful to document the format being parsed/have examples of valid values